saveHandler.js

```diff
@@ -0,0 +1,52 @@
+var _ = require('lodash');
+var saveCommand = "$save";
+var commands;
+
+
+exports.isSaveCommand = function (msg) {
+	if(msg.indexOf(saveCommand) > -1) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+exports.setCommands = function (allowedCommands) {
+	commands = new Commands();
+	_.forEach(allowedCommands, function(obj) {
+		commands.addCommand(obj);
+	});
+}
+
+exports.handleSaveCommand = function(saveRequest) {
+	var msg = saveRequest.text;
+	var array  = msg.split("-");
+	array.shift();
+	_.forEach(array, function(strValue) {
+		var arguments = strValue.split(" ");
+		var command = commands.getCommand(arguments[0]); 
+		if(!command) {
+			return;
+		}
+		arguments.shift();
+		command.handler(arguments, saveRequest);
+	})
+}
+
+var Commands = function() {
```

Comment:
Flytt ``Commands`` til en egen modul. Og hvis du skal bruke prototyp-objekter foretrekker jeg at du bruker ``function Commands`` for deklarering istedenfor.